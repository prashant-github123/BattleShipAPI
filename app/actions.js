import request from 'axios';
import {
  ADD_PLAYER,
  SHIP_COORDS,
  SHIP_LOCATION,
  STATUS,
    ADD_NEW_USER_URI,
    PLACE_SHIP_URI,
    RETRIVE_SHIP_LOC_URI,
    CHECK_TURN_URI
 } from './constants';
import {browserHistory} from 'react-router';

export const addPlayerAPI = (playerName) => { // eslint-disable-line
  console.log(playerName);
  const data = {
    playerName: playerName,
  }
  return request({
    method: 'POST',
    url: ADD_NEW_USER_URI,
    data,
  });
};

export const retrieveShipLocationsAPI = (gameId) => { // eslint-disable-line
  const data = {
    gameId : gameId,
  }
  return request({
    method: 'POST',
    url: RETRIVE_SHIP_LOC_URI,
    data,
  });
};

export const placeShipAPI = (playerData, cords) => { // eslint-disable-line
  console.log(playerData);
  const data = {
    shipCoordinates: cords.toString(),
    gameId: playerData.gameID,
    playerId: playerData.playerID,
    playerName: playerData.playerName,
  };
  return request({
    method: 'POST',
    url: PLACE_SHIP_URI,
    data,
  });
};

export const checkTurnStatusAPI = (gamedata) => {
  console.log(gamedata);
  return request({
    method: 'POST',
    url: CHECK_TURN_URI,
    data: gamedata,
  });
};

export const setCurrentUser = (data) => {
  return {
    type: ADD_PLAYER,
    data,
  };
};

export const setShipCoordinates = (data) => { // eslint-disable-line
  return {
    type: SHIP_COORDS,
    data,
  };
};

export const setShipLocation = (data) => { // eslint-disable-line
  return {
    type: SHIP_LOCATION,
    data,
  };
};

export const setTurnStatus = (data) => { // eslint-disable-line
  return {
    type: STATUS,
    data,
  };
};

/*
* Function addPlayer
* @param : playerName
*/
export const addPlayer = (playerName) => { //eslint-disable-line
  return function (dispatch) {
    addPlayerAPI(playerName).then((resp) => { // eslint-disable-line
      console.log(resp);
      dispatch(setCurrentUser(resp.data));
      window.location.href = '/placeShip';
    }).catch((err) => {
      Error(err);
    });
  };
};
/*
* Function placeShip
* @param : playerData : Player Information
* @param : Cords : Ship Co-ordinates
*/
export const placeShip = (playerData, cords) => { // eslint-disable-line
  console.log(playerData, cords);
  return function () {
    placeShipAPI(playerData, cords).then((resp) => { // eslint-disable-line
      window.location.href = '/gameBoard';
    }).catch((err) => {
      Error(err);
    });
  };
};
/*
* Function retriveShipLocation
* @param : gameId
*/
export const retriveShipLocation = (gameId) => {// eslint-disable-line
  return function (dispatch) {
    retrieveShipLocationsAPI(gameId).then((response) => {
      dispatch(setShipLocation(response.data));
      return response.data;
    }).catch((err) => {
      console.log("Error", err);
    });
  };
};
/*
* Function checkTurnStatus
* @param : data
*/
export const checkMyTurnStatus = (data) => { // eslint-disable-line

  return checkTurnStatusAPI(data).then((response) => {
    return response.data;
  }).catch((err) => {
    console.log("Error", err);
  });
};

/*Start New Game*/
export const startNew = () => {
  sessionStorage.clear();
  browserHistory.push('/');
}
