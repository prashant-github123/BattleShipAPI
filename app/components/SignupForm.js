import React, { Component, PropTypes } from 'react';

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      playerName: '',
      status: false,
    };
  }

  addUserName = (e) => {
    const val = e.target.value;
    this.setState({
      playerName: val,
    });
  }
  handleSubmit = () => {
    const { playerName } = this.state;
    const { addPlayer } = this.props;
    addPlayer(playerName);
  }

  render() {
    return (
      <div className="App">
        <div className="user-box clearboth">
          <div className="user-one float-left">
            <div className="player-title">Enter your Details</div>
            <div>
              <input id="playerName" onChange={this.addUserName} type="text" placeholder="Please Enter name" />
            </div>
            <button onClick={this.handleSubmit}>submit</button>
          </div>
        </div>
      </div>
    );
  }
}
SignupForm.defaultProps = {
  addPlayer: function addPlayer() {
    return {};
  },
};

SignupForm.propTypes = {
  addPlayer: PropTypes.func,
};

export default SignupForm;
