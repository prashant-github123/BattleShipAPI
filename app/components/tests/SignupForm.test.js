import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SignupForm from './../SignupForm';
import { addPlayer } from './../../actions';
// Snapshot for SignUpForm React Component
describe('Sign UP Form --- Snapshot', () => {
  it('capturing Snapshot of SignUPForm', () => {
    const renderedValue = renderer.create(<SignupForm addPlayer={addPlayer} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

//* ******************************************************************************************************
describe('Sign Up Form --- Shallow Render REACT COMPONENTS', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignupForm addPlayer={addPlayer} />);
  });

  it('should render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('should contains button ', () => {
    expect(wrapper.find('button').type()).toEqual('button');
  });
  it('should contains input ', () => {
    expect(wrapper.find('input').type()).toEqual('input');
  });
  it('should call addPlayer on click of input value', () => {
    wrapper.addPlayer = jest.fn();
    wrapper.find('button').simulate('click');
    wrapper.addPlayer(); // check this again
    expect(wrapper.addPlayer).toHaveBeenCalled();
  });
  it('should call addPlayer on Change of input value', () => {
    wrapper.find('input').simulate('change', { target: {
      value: 'Player1',
    } });
    expect(wrapper.state('playerName')).toEqual('Player1');
  });
});
