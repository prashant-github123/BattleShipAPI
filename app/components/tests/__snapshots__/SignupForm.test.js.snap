// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign UP Form --- Snapshot capturing Snapshot of SignUPForm 1`] = `
<div
  className="App"
>
  <div
    className="user-box clearboth"
  >
    <div
      className="user-one float-left"
    >
      <div
        className="player-title"
      >
        Enter your Details
      </div>
      <div>
        <input
          id="playerName"
          onChange={[Function]}
          placeholder="Please Enter name"
          type="text"
        />
      </div>
      <button
        onClick={[Function]}
      >
        submit
      </button>
    </div>
  </div>
</div>
`;
